cupy/core/core.pyx

```diff
@@ -666,7 +671,47 @@ cdef class ndarray:
 
         return out
 
-    # TODO(okuta): Implement sort
+    def sort(self):
+        """Sort an array, in-place with a stable sorting algorithm.
+
+        .. note::
+           For its implementation reason, ``ndarray.sort`` currently supports
+           only arrays with their rank of one and their own data, and does not
+           support ``axis``, ``kind`` and ``order`` parameters that
+           ``numpy.ndarray.sort`` does support.
+
+        .. seealso::
+            :func:`cupy.sort` for full documentation,
+            :meth:`numpy.ndarray.sort`
+
+        """
+
+        # TODO(takagi): Support axis argument.
+        # TODO(takagi): Support kind argument.
+
+        if self.shape == ():
+            msg = 'Sorting arrays with the rank of zero is not supported'
+            raise ValueError(msg)
+
+        # TODO(takagi): Support ranks of two or more
+        if len(self.shape) > 1:
+            msg = ('Sorting arrays with the rank of two or more is '
+                   'not supported')
+            raise ValueError(msg)
+
+        # TODO(takagi): Support sorting views
+        if self.base is not None:
```

Comment:
It is currently assumed that a sorted array always has rank one, which is ensured C contiguous as well as F contiguous, so checking c_contiguous would not be required this time.